src/main/java/carpet/mixins/ServerGamePacketListenerImplMixin.java

```diff
@@ -24,6 +26,10 @@ private void onCustomCarpetPayload(ServerboundCustomPayloadPacket packet, Callba
         ResourceLocation channel = packet.getIdentifier();
         if (CarpetClient.CARPET_CHANNEL.equals(channel))
         {
+            // We should force onto the main thread here
+            // ServerNetworkHandler.handleData can possibly mutate data that isn't
+            // thread safe, and also allows for client commands to be executed
+            PacketUtils.ensureRunningOnSameThread(packet, (ServerGamePacketListener) this, player.getLevel());
```

Comment:
@altrisi do you know why you needed to .copy() here? seems only carpet is using these packets

Reply:
I don't know, this is what was previously implemented in the mixin with this behavior, I just inlined it when the mixin was no longer needed.

It was previously here:
https://github.com/gnembon/fabric-carpet/blob/19ad8cfbf3a0d92eb0d133eaf4fb15bf808034d7/src/main/java/carpet/mixins/CustomPayloadC2SPacketMixin.java#L25

Reply:
I assume it might be a result of the packet buf being released at some point and causing issues?
I'm pretty sure that it's redundant now.